src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -121,36 +153,66 @@ CHIP_ERROR PairingSession::DecodeMRPParametersIfPresent(TLV::Tag expectedTag, TL
     if (TLV::TagNumFromTag(tlvReader.GetTag()) == 1)
     {
         ReturnErrorOnFailure(tlvReader.Get(tlvElementValue));
-        mRemoteMRPConfig.mIdleRetransTimeout = System::Clock::Milliseconds32(tlvElementValue);
+        mRemoteSessionParam.SetMRPIdleRetransTimeout(System::Clock::Milliseconds32(tlvElementValue));
 
         // The next element is optional. If it's not present, return CHIP_NO_ERROR.
-        CHIP_ERROR err = tlvReader.Next();
-        if (err == CHIP_END_OF_TLV)
-        {
-            return tlvReader.ExitContainer(containerType);
-        }
-        ReturnErrorOnFailure(err);
+        IncrementReaderExitOnEndOfTlv(tlvReader);
     }
 
     if (TLV::TagNumFromTag(tlvReader.GetTag()) == 2)
     {
         ReturnErrorOnFailure(tlvReader.Get(tlvElementValue));
-        mRemoteMRPConfig.mActiveRetransTimeout = System::Clock::Milliseconds32(tlvElementValue);
+        mRemoteSessionParam.SetMRPActiveRetransTimeout(System::Clock::Milliseconds32(tlvElementValue));
 
         // The next element is optional. If it's not present, return CHIP_NO_ERROR.
-        CHIP_ERROR err = tlvReader.Next();
-        if (err == CHIP_END_OF_TLV)
-        {
-            return tlvReader.ExitContainer(containerType);
-        }
-        ReturnErrorOnFailure(err);
+        IncrementReaderExitOnEndOfTlv(tlvReader);
     }
 
     if (TLV::TagNumFromTag(tlvReader.GetTag()) == 3)
     {
         ReturnErrorOnFailure(tlvReader.Get(tlvElementValue));
-        mRemoteMRPConfig.mActiveThresholdTime = System::Clock::Milliseconds16(tlvElementValue);
+        mRemoteSessionParam.SetMRPActiveThresholdTime(System::Clock::Milliseconds16(tlvElementValue));
+
+        // The next element is optional. If it's not present, return CHIP_NO_ERROR.
+        IncrementReaderExitOnEndOfTlv(tlvReader);
+    }
+
+    if (TLV::TagNumFromTag(tlvReader.GetTag()) == 4)
```

Comment:
should we add constants for these structures? with some comment that these are from `4.12.8 Parameters and Constants` ?